### PR TITLE
Crud methods

### PR DESCRIPTION
**Pull request by wesonweb**
# Complete CRUD operations

- Completes the remaining CRUD actions. 
- updates the model to add some enun values 
- changes one endpoint for POST
- merges main into this branch

## Plant structure

Screenshot of how the plant object is structured so far. Have tested all end points using Postman and all return OK status.
Screenshot 2023-09-22 at 14 58 15

# Client side setup

- installs Vite, tailwind and other dependencies
- sets up react-router-dom and creates a layout and route to home page

### CODE REVIEW

File: server/routes/plants.js [hunk 1]
```diff
@@ -1,25 +1,19 @@
...
 
 // POST a single plant (create)
-router.post('/', createPlant)
+router.post('/new-plant', createPlant)
 
 // DELETE a single plant
-router.delete('/:id', (req, res) => {
-  res.json({message: 'DELETE one plant'})
-})
+router.delete('/:id', deletePlant)
```
Comment (lc512k) [inline server/routes/plants.js:15]: Nicely done with these - as you have more functionality it pays off to have things tucked away in their own files

File: server/models/PlantModel.js [hunk 2]
```diff
@@ -51,7 +52,13 @@ const PlantSchema = new Schema({
     },
     problems: {
       type: [String],
+    },
+    care_level: {
```
Comment (lc512k) [inline server/models/PlantModel.js:56]: Don't go by me too much cos it's been a while, but I used to do camelCase with mongo too. Not sure if I was doing it right

File: server/controllers/plantController.js [hunk 3]
```diff
@@ -46,19 +48,60 @@ const getPlants = async (req, res) => {
...
 // delete a plant by id
 
+const deletePlant = async (req, res) => {
+  const { id } = req.params
+  try {
+    const plant = await Plant.findByIdAndDelete({_id: id})
+    if (!plant) {
+      return res.status(400).json({message: `could not find plant`})
+    }
+    return res.status(200).json({message: `plant deleted`})
```
Comment (lc512k) [inline server/controllers/plantController.js:93]: nice use of response code

File: client/src/App.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import {  // only use for react-router-dom v6 and above
```
Comment (lc512k) [inline client/src/App.jsx:1]: for the React code I would advise having someone who knows some react give you feedback. I can read it and it looks good to me, but I did very little React in my coding days :|

**Review by lc512k — APPROVED** [review]
Nice work!

File: server/models/PlantModel.js (hunk 2 above)
Comment (wesonweb) [reply server/models/PlantModel.js:56]: Good point. When I did it I wondered if camelCase would be better for consistency - I'll change it :)

**Pull request merged**